### PR TITLE
Solucao provisoria parao social recycler view

### PR DESCRIPTION
**Pull request by jbalves**
@gabeira @firemanxbr @DanielNogueiraAndroid @gleidesigner @Wandersonjack

### CODE REVIEW

File: app/src/main/java/mobi/plantare/fragments/SocialFragment.java [hunk 1]
```diff
@@ -64,10 +87,95 @@ public void onCreate(Bundle savedInstanceState) {
...
+            @Override
+            public void onCancelled(DatabaseError error) {
+                //Failed to read value
+                Log.w(TAG,"Failed to read value.",error.toException());
+            }
+        });
+
+        return lista;
+
+        /*
```
Comment (gabeira) [inline app/src/main/java/mobi/plantare/fragments/SocialFragment.java]: Remove comments

File: app/src/main/java/mobi/plantare/fragments/SocialFragment.java [hunk 2]
```diff
@@ -64,10 +87,95 @@ public void onCreate(Bundle savedInstanceState) {
...
+                Log.e(TAG,"Size: " + dataSnapshot.getChildrenCount());
+                for (DataSnapshot dataSnap : dataSnapshot.getChildren()){
+                    Plant plant = dataSnap.getValue(Plant.class);
+                    //mclusterManager.addItem(plant);
+                    lista.add(plant);
+                    Log.e(TAG,"Plant (name): " + plant.getName());
+                    Log.e(TAG,"Plant (type): " + plant.getType());
+                    Log.e(TAG,"Plant (photo): " + plant.getPhoto());
+                }
+                //mclusterManager.cluster();
```
Comment (gabeira) [inline app/src/main/java/mobi/plantare/fragments/SocialFragment.java]: Add 
`adapter.notifyDataSetChanged();`

File: app/src/main/java/mobi/plantare/fragments/SocialFragment.java [hunk 3]
```diff
@@ -64,10 +87,95 @@ public void onCreate(Bundle savedInstanceState) {
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
+
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_social, container, false);
+        View layout = inflater.inflate(R.layout.fragment_social, container, false);
+
+        ArrayList<Plant> lista = getPlants();
```
Comment (gabeira) [inline app/src/main/java/mobi/plantare/fragments/SocialFragment.java]: Declare as a class variable and use once.

File: app/src/main/java/mobi/plantare/adapters/SocialListAdapter.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        }
+
+        public ViewHolder setImg(String imageBytes){
+            if (imgPlantView == null) return this;
+
+            //Convert from string to Bitmap
+            //byte[] imageByteArray = Base64.decode(imageBytes, Base64.DEFAULT);
+            //Bitmap bmp = BitmapFactory.decodeByteArray(byteArray, 0, byteArray.length);
+
+            /* Ainda não está funcionando o set de imagens das plantas
```
Comment (gabeira) [inline app/src/main/java/mobi/plantare/adapters/SocialListAdapter.java]: Forget Glide here, you already have the bitmap, just load it.
`imgPlantView.setImageBitmap(bmp);`

File: app/src/main/java/mobi/plantare/adapters/SocialListAdapter.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        }
+
+        public ViewHolder setDescription(String description){
+            if (descriptionPlantView == null) return this;
+            descriptionPlantView.setText(description);
+            return this;
+        }
+
+        public ViewHolder setImg(String imageBytes){
+            if (imgPlantView == null) return this;
```
Comment (gabeira) [inline app/src/main/java/mobi/plantare/adapters/SocialListAdapter.java:101]: Check also for imageBytes == null
Than load another placeholder image like iplanted2

File: app/src/main/res/layout/item_social_list.xml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:weightSum="1">
+
+    <ImageView
+        android:id="@+id/imgPlant"
+        android:paddingLeft="16dp"
+        android:layout_width="60dp"
+        android:layout_height="60dp"
+        android:background="@mipmap/ic_launcher"/>
```
Comment (gabeira) [inline app/src/main/res/layout/item_social_list.xml]: Remove background, rather will have both images
Maybe just another color meaning that will load an image there.

File: app/src/main/res/layout/item_social_list.xml [hunk 7]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:weightSum="1">
+
+    <ImageView
```
Comment (gabeira) [inline app/src/main/res/layout/item_social_list.xml:8]: Will be nice to set an scaleType as well.

**Review by gabeira — CHANGES_REQUESTED** [review]
Also remember to reformat the code
Menu>Code>Reformat Code
To keep it better organized.
And congratulations for the feature.

**Review by gabeira — APPROVED** [review]

**Review by gleideveloper — APPROVED** [review]
Done

**Pull request merged**